### PR TITLE
[Kodi] Added new channel to play a favorite

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added new channel `playfavorite` to play a Kodi favorite (state options are filled dynamically)
- Use `refreshInterval` for both jobs

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

**Comment by cweitkamp** [comment]
@mhilbush Do you have time for an initial review? Would be nice.

**Comment by mhilbush** [comment]
@cweitkamp Sure thing. I'll be a little pressed for time today, but I can look at it tomorrow and/or Monday.

File: addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/model/KodiFavorite.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+     */
+    private String windowParameter;
+
+    public String getTitle() {
+        return title;
+    }
+
+    public void setTitle(final String title) {
+        Matcher m = TITLE_PATTERN.matcher(title);
+        this.title = m.replaceAll("");
```
[the remark this answers is not in the document]
Comment (cweitkamp) [reply addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/model/KodiFavorite.java]: See example and description in detail: https://docs.oracle.com/javase/tutorial/essential/regex/matcher.html - Using replaceFirst(String) and replaceAll(String).

**Comment by cweitkamp** [comment]
@mhilbush Sounds good. Thank you very much.

File: addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/model/KodiFavorite.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (mhilbush) [reply addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/model/KodiFavorite.java]: Ah, ok. Wasn't familiar with that.

**Comment by cweitkamp** [comment]
@martinvw may I ask you to have a look at this PR? It's already reviewed. Thank you.

File: addons/binding/org.openhab.binding.kodi/ESH-INF/thing/thing-types.xml [hunk 2]
```diff
@@ -66,16 +67,24 @@
 		<label>Play URI</label>
 		<description>Play the given URI</description>
 	</channel-type>
+	<channel-type id="playfavorite">
+		<item-type>String</item-type>
+		<label>Play a Favorite</label>
+		<description>Play the given favorite by sending a command with the favorite's title</description>
+		<state pattern="%s" />
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.kodi/ESH-INF/thing/thing-types.xml]: Isn't this already the default?

File: addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/handler/KodiHandler.java [hunk 3]
```diff
@@ -162,6 +163,13 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                     updateState(CHANNEL_PLAYURI, UnDefType.UNDEF);
                 }
                 break;
+            case CHANNEL_PLAYFAVORITE:
+                if (command instanceof StringType) {
+                    playFavorite(command);
+                } else if (command.equals(RefreshType.REFRESH)) {
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/handler/KodiHandler.java]: AFAIK `RefreshType` is an enum so you could use == is I'm correct.

**Comment by martinvw** [comment]
If you want you can fix them otherwise I'll just merge :-)

**Comment by cweitkamp** [comment]
I changed it. Thank you.

**Pull request merged**